controllers/awsmachine_controller.go

```diff
@@ -796,6 +801,29 @@ func (r *AWSMachineReconciler) getInfraCluster(ctx context.Context, log logr.Log
 		return managedControlPlaneScope, nil
 	}
 
+	if cluster.Spec.InfrastructureRef != nil &&
+		cluster.Spec.InfrastructureRef.Kind != AWSClusterInfrastructureRefKind {
+		externalInfraCluster, err := external.Get(ctx, r.Client, cluster.Spec.InfrastructureRef, cluster.Spec.InfrastructureRef.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching externalInfraCluster: %w", err)
```

Comment:
```suggestion
			log.Error(err, "error fetching externalInfraCluster")
			return nil, err
```
Doing this means we get the source code line rather than it pointing in controller runtime. Will make it easier to debug if there's a problem.